numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,18 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    def __new__(cls, pyfunc=None, *args, **kw):
```

Comment:
hm, maybe this could bear to have a comment justifying *why* we are resorting to overriding `__new__` (namely, backwards compatibility)